sync/syncer.go

```diff
@@ -198,89 +216,191 @@ func (s *Syncer) getLayerFromNeighbors(currenSyncLayer types.LayerID) (*types.La
 	return types.NewExistingLayer(types.LayerID(currenSyncLayer), blocksArr), nil
 }
 
-func (s *Syncer) fetchFullBlocks(blockIds []types.BlockID) ([]*types.Block, error) {
-	output := s.fetchWithFactory(BlockReqFactory(blockIds), s.Concurrency)
+func (s *Syncer) FetchFullBlocks(blockIds []types.BlockID) ([]*types.Block, error) {
+	output := s.fetchWithFactory(NewBlockWorker(s, s.Concurrency, BlockReqFactory(), blockSliceToChan(blockIds)))
 	blocksArr := make([]*types.Block, 0, len(blockIds))
 	for out := range output {
 		block := out.(*types.Block)
-
-		eligible, err := s.BlockEligible(&block.BlockHeader)
-		if err != nil {
-			s.Warning(fmt.Sprintf("failed checking eligiblety %v", block.ID()), err)
-			continue
-		}
-		if !eligible {
-			s.Warning(fmt.Sprintf("block %v not eligible", block.ID()), err)
-			continue
-		}
-
-		if err := s.syncMissingContent(block); err != nil {
-			s.Error(fmt.Sprintf("failed derefrencing block data %v", block.ID()), err)
+		if err := s.SyncAndValidate(block); err != nil {
+			s.Error(fmt.Sprintf("failed derefrencing block data %v %v", block.ID(), err))
 			continue
 		}
 
 		s.Info("added block to layer %v", block.Layer())
```

Comment:
Consider counting the number of blocks that were added and replace this log print with a single print with the final count (in order to reduce the logs sent to Elastic)

Reply:
we want the id of the problem block for debugging maybe when we are more stable we will loose this